controllers/openshift_controller.go

```diff
@@ -1833,6 +1842,163 @@ func (r *KataConfigOpenShiftReconciler) updateStatus() error {
+	return nodeMcoState == NodeDegraded && isKataEnabledOnNode
+}
+
+func isNodeFailedToUninstall(nodeMcoState string, nodeCurrMc string, nodeTargetMc string, isKataEnabledOnNode bool) bool {
+	return nodeMcoState == NodeDegraded && !isKataEnabledOnNode
+}
+
+func (r *KataConfigOpenShiftReconciler) putNodeOnStatusList(node *corev1.Node) error {
+
+	isConvergedCluster, err := r.checkConvergedCluster()
+	if err != nil {
+		return err
+	}
+
+	targetMcpName := func() string {
+		if isConvergedCluster {
+			return "master"
+		}
+		_, nodeLabeledForKata := node.Labels["node-role.kubernetes.io/kata-oc"]
+		if nodeLabeledForKata {
+			return "kata-oc"
+		} else {
+			return "worker"
+		}
+	}()
+
+	targetMcp, err := r.getMcpByName(targetMcpName)
+	if err != nil {
+		return err
+	}
+
+
+	nodeMcoState, ok := node.Annotations["machineconfiguration.openshift.io/state"]
+	if !ok {
+		return fmt.Errorf("Missing machineconfiguration.openshift.io/state on node %v", node.GetName())
+	}
+
+	nodeCurrMc, ok := node.Annotations["machineconfiguration.openshift.io/currentConfig"]
+	if !ok {
+		return fmt.Errorf("Missing machineconfiguration.openshift.io/currentConfig on node %v", node.GetName())
+	}
+
+	// Note that to figure out the MachineConfig our Node should be at we
+	// unfortunately cannot use
+	// machineconfiguration.openshift.io/desiredConfig as would seem
+	// logical and easy.  The reason is that the MCO only sets
+	// `desiredConfig` to the actual desired config right before it starts
+	// updating the Node.  So to get the correct target MachineConfig we
+	// need to look at the MachineConfigPool that our Node belongs to or
+	// will belong to shortly.
+	nodeTargetMc := targetMcp.Spec.Configuration.Name
+
+	// `isKataEnabledOnNode` is a per Node condition on regular clusters
+	// but cluster-wide on converged ones.
+	// On regular clusters, this is ultimately determined by
+	// KataConfig.spec.kataConfigPoolSelector (we use the
+	// node-role.kubernetes.io/kata-oc to find this above in this function,
+	// and the node-role is in turn assigned to Nodes based on the pool
+	// selector).
+	// On converged clusters, basically only two operations are possible:
+	// installing kata on all masters and uninstalling kata from all
+	// masters, no per-Node options can be supported.  We find if kata is
+	// supposed to be installed on the cluster by examining the "master"
+	// MCP's MachineConfig to see if it installs the kata containers
+	// extension.
+	var isKataEnabledOnNode bool
+	if isConvergedCluster {
+		targetMc := &mcfgv1.MachineConfig{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: targetMcp.Spec.Configuration.Name}, targetMc)
+		if err != nil {
+			r.Log.Info("Failed to retrieve MachineConfig", "MC name", targetMcp.Spec.Configuration.Name, targetMc, "MCP name", targetMcpName)
+			return err
+		}
+
+		isKataEnabledOnNode = func () bool {
```

Comment:
Suggested by `go fmt ./...` :

```suggestion
		isKataEnabledOnNode = func() bool {
```

Reply:
Fixed, thanks!